WearComplicationProvidersTestSuite/Wearable/src/main/java/com/example/android/wearable/wear/wearcomplicationproviderstestsuite/RangedValueProviderService.kt

```diff
@@ -54,46 +59,115 @@ class RangedValueProviderService : SuspendingComplicationProviderService() {
             )
         // Suspending function to retrieve the complication's state
         val state = args.getState(this)
-        val caseValue = state.mod(4)
-        val minValue = MIN_VALUES[caseValue]
-        val maxValue = MAX_VALUES[caseValue]
+        val case = Case.values()[state.mod(Case.values().size)]
+
+        return getComplicationData(
+            tapAction = complicationTogglePendingIntent,
+            case = case
+        )
+    }
+
+    override fun getPreviewData(type: ComplicationType): ComplicationData? =
+        getComplicationData(
+            tapAction = null,
+            case = Case.TEXT_WITH_ICON
+        )
+
+    private fun getComplicationData(
+        tapAction: PendingIntent?,
+        case: Case
+    ): ComplicationData {
+        val text: ComplicationText?
+        val monochromaticImage: MonochromaticImage?
+        val title: ComplicationText?
+        val caseContentDescription: String
+
+        val minValue = case.minValue
+        val maxValue = case.maxValue
         val value = Random.nextDouble(minValue.toDouble(), maxValue.toDouble()).toFloat()
+        val percentage = (value - minValue) / (maxValue - minValue)
+
+        when (case) {
+            Case.TEXT_ONLY -> {
+                text = PlainComplicationText.Builder(
```

Comment:
Ditto here.

Reply:
I left `RangedValueProviderService` as is, due to the `contentDescription` logic that is dependent on the case.